controllers/installation_controller.go

```diff
@@ -597,6 +598,28 @@ func (r *InstallationReconciler) ReconcileK0sVersion(ctx context.Context, in *v1
 	return nil
 }
 
+// ReconcileRegistry reconciles registry components, ensuring that the necessary secrets are
+// created as well as rebalancing stateful pods when nodes are removed from the cluster.
+func (r *InstallationReconciler) ReconcileRegistry(ctx context.Context, in *v1beta1.Installation) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	metadata, err := release.MetadataFor(ctx, in, r.Client)
+	if err != nil {
+		in.Status.SetState(v1beta1.InstallationStateHelmChartUpdateFailure, err.Error(), nil)
```

Comment:
I'm not sure that this is the perfect state name, but I'm not sure what it should be either

Reply:
hah, and it's what we're using elsewhere
ignore me